src/components/CreateCard/UpdateCard.js

```diff
@@ -0,0 +1,136 @@
+import React, {useState} from "react";
+import {Form, Icon, Input, Button, Spin, Alert} from "antd";
+import "../CreateCard/AddCard.module.css";
+import {axiosWithAuth} from "../../utils/axios";
+import {updateCard} from "../../state/actions/CardAction";
+import {connect} from "react-redux";
+import styles from "./AddCard.module.css";
+
+function UpdateCard(props) {
+  const [formValues, setFormValues] = useState({
+    deckId: "",
+    questionText: "",
+    answerText: "",
+  });
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const {getFieldDecorator} = props.form;
+
+  const handleChange = e => {
+    setFormValues({
+      ...formValues,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = e => {
+    console.log(props);
+    e.preventDefault();
+    const newCard = {
+      deckId: props.currentDeckState.deck_id,
+      questionText: formValues.questionText,
+      answerText: formValues.answerText,
+    };
+    console.log(newCard);
+
+    setLoading(true);
+    console.log(props.id);
+
+    props.updateCard(props.cardId, newCard);
+    props.toggleModal();
+    // Call the server after authentication
+    // // axiosWithAuth()
+    // //   .put(`/cards/${props.location.state.id}`, newCard)
+    // //   .then(res => {
+    setLoading(false);
```

Comment:
Does this actually work for displaying a loading state? Looks like you are not handling the asyncronisity here so it would not wait for things to resolve before it sets loading to false.